openassessment/xblock/submission_mixin.py

```diff
@@ -410,34 +410,59 @@ def upload_url(self, data, suffix=''):  # pylint: disable=unused-argument
             A URL to be used to upload content associated with this submission.
 
         """
+        # Validate that there are no data issues and file type is allowed
+        try:
+            if not self.is_supported_upload_type(data):
+                return {'success': False, 'msg': self._(
+                    "File upload failed: unsupported file type."
+                    "Only the supported file types can be uploaded."
+                    "If you have questions, please reach out to the course team."
+                )}
+        except KeyError:
+            return {'success': False, 'msg': self._("There was an error uploading your file.")}
+
+        # Attempt to upload
+        file_num = int(data.get('filenum', 0))
+        content_type = data['contentType']
+        try:
+            key = self._get_student_item_key(file_num)
+            url = file_upload_api.get_upload_url(key, content_type)
+            return {'success': True, 'url': url}
+        except FileUploadError:
+            logger.exception("FileUploadError:Error retrieving upload URL for the data:{data}.".format(data=data))
+            return {'success': False, 'msg': self._("Error retrieving upload URL.")}
+
+    def is_supported_upload_type(self, data):
+        """
+        Determine if the uploaded file type is allowed for our upload configuration
+
+        Returns:
+            True/False if file type is supported/unsupported
+
+        Raises:
+            KeyError when upload data is missing or incomplete
+        """
         if 'contentType' not in data or 'filename' not in data:
-            return {'success': False, 'msg': self._(u"There was an error uploading your file.")}
+            raise KeyError()
+
         content_type = data['contentType']
         file_name = data['filename']
         file_name_parts = file_name.split('.')
-
-        file_num = int(data.get('filenum', 0))
-
         file_ext = file_name_parts[-1] if len(file_name_parts) > 1 else None
```

Comment:
I know that this is the existing behavior, but is there a more robust API for dealing with file types than just doing string split on periods?

Reply:
Yes. Have updated using `os.splitext()`